internal/actions/release.go

```diff
@@ -146,3 +155,57 @@ func addPublishOutputs(dir string, outputs map[string]string) error {
 
 	return nil
 }
+
+func addCurrentBranchTagging(g *git.Git, latestRelease map[string]releases.LanguageReleaseInfo) error {
+	_, err := cli.Download("latest", g)
+	if err != nil {
+		return err
+	}
+
+	var sources, targets []string
+	branch := strings.TrimPrefix(os.Getenv("GITHUB_REF"), "refs/heads/")
+	workflow, err := configuration.GetWorkflowAndValidateLanguages(true)
+	if err != nil {
+		return err
+	}
+
+	if specificTarget := environment.SpecifiedTarget(); specificTarget != "" {
+		if target, ok := workflow.Targets[specificTarget]; ok {
+			sources = append(sources, target.Source)
+			targets = append(targets, specificTarget)
+		}
+	} else {
+		for name, target := range workflow.Targets {
+			if releaseInfo, ok := latestRelease[target.Target]; ok {
+				releasePath, err := filepath.Rel(".", releaseInfo.Path)
+				if err != nil {
+					return err
+				}
+
+				// check for no SDK output path
+				if (releasePath == "" || releasePath == ".") && target.Output == nil {
+					sources = append(sources, target.Source)
+					targets = append(targets, name)
+				}
+
+				if target.Output != nil {
+					outputPath, err := filepath.Rel(".", *target.Output)
+					if err != nil {
+						return err
+					}
+					outputPath = filepath.Join(environment.GetWorkingDirectory(), outputPath)
+					if outputPath == releasePath {
+						sources = append(sources, target.Source)
+						targets = append(targets, name)
+					}
+				}
+			}
+		}
+	}
+
+	if len(sources) > 0 && len(targets) > 0 && branch != "" {
```

Comment:
Why do we need sources AND targets?

Reply:
It's just weird naming with the way the underlying tagging lib is currently written. Sources is basically the spec source and targets is codeSamples. I'll leave a comment here